src/services/exit_order.py

```diff
@@ -0,0 +1,360 @@
+        operator_state: NodeOperatorPredictableState,
+        total_predictable_validators_count: int,
+    ) -> int:
+        stake_volume = 100 * operator_state.predictable_validators_count / total_predictable_validators_count
+        stake_volume_weight = operator_state.predictable_validators_total_age if stake_volume > 1 else 0
+        return stake_volume_weight
+
+    @staticmethod
+    def _operator_predictable_validators(operator_state: NodeOperatorPredictableState) -> int:
+        return operator_state.predictable_validators_count
+
+    @staticmethod
+    def _validator_index(validator: LidoValidator) -> int:
+        return int(validator.index)
+
+    def _decrease_node_operator_stats(self, validator: LidoValidator) -> None:
+        """
+        Sub particular validator stats from its node operator stats
+        We do it every time when validator is popped from the queue for resort the rest of queue
+        """
+        module_operator = self._no_index_by_validator(validator)
+        self.total_predictable_validators_count -= 1
+        self.lido_node_operator_stats[module_operator].predictable_validators_count -= 1
+        self.lido_node_operator_stats[module_operator].predictable_validators_total_age -= max(
+            0, self.blockstamp.ref_epoch - int(validator.validator.activation_epoch)
+        )
+
+    #  -- Internal methods to interact with operators state for sorting --
+    def _get_config(self):
+        orl = named_tuple_to_dataclass(
+            self.w3.lido_contracts.oracle_report_sanity_checker.functions.getOracleReportLimits().call(
+                block_identifier=self.blockstamp.block_hash,
+            ),
+            OracleReportLimits
+        )
+        validator_delayed_timeout_in_slots = Web3.to_int(
+            self.w3.lido_contracts.oracle_daemon_config.functions.get('VALIDATOR_DELAYED_TIMEOUT_IN_SLOTS').call(
+                block_identifier=self.blockstamp.block_hash,
+            )
+        )
+        self.v_conf = ValidatorToExitIteratorConfig(
+            orl.max_validator_exit_requests_per_report,
+            validator_delayed_timeout_in_slots
+        )
+
+    def _get_exitable_lido_validators(
+        self,
+        operator_validators: ValidatorsByNodeOperator,
+        last_requested_to_exit_indices_per_operator: dict[NodeOperatorGlobalIndex, int]
+    ) -> Iterable[LidoValidator]:
+        for module_operator, validators in operator_validators.items():
+            for v in validators:
+                # https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/beacon-chain.md#voluntary-exits
+                requested_to_exit = (
+                    int(v.index) <= last_requested_to_exit_indices_per_operator[module_operator]
+                )
+                on_exit = self._is_on_exit(v)
+                if not on_exit and not requested_to_exit:
+                    yield v
+
+    def _no_index_by_validator(self, validator: LidoValidator) -> NodeOperatorGlobalIndex:
```

Comment:
please consider to rename this one (see above)

Reply:
will be fixed during refactoring